homeassistant/components/braviatv/__init__.py

```diff
@@ -44,3 +67,223 @@ async def async_unload_entry(hass, config_entry):
+                return
+
+        self.is_on = False
+
+    async def _async_update_data(self):
+        """Fetch the latest data."""
+        if self.state_lock.locked():
+            return
+
+        await self.hass.async_add_executor_job(self._update_tv_data)
+
+    async def async_turn_on(self):
+        """Turn the device on."""
+        async with self.state_lock:
+            await self.hass.async_add_executor_job(self.braviarc.turn_on)
+            await self.async_request_refresh()
+
+    async def async_turn_off(self):
+        """Turn off device."""
+        async with self.state_lock:
+            await self.hass.async_add_executor_job(self.braviarc.turn_off)
+            await self.async_request_refresh()
+
+    async def async_set_volume_level(self, volume):
+        """Set volume level, range 0..1."""
+        async with self.state_lock:
+            await self.hass.async_add_executor_job(
+                self.braviarc.set_volume_level, volume, self.audio_output
+            )
+            await self.async_request_refresh()
+
+    async def async_volume_up(self):
+        """Send volume up command to device."""
+        async with self.state_lock:
+            await self.hass.async_add_executor_job(
+                self.braviarc.volume_up, self.audio_output
+            )
+            await self.async_request_refresh()
+
+    async def async_volume_down(self):
+        """Send volume down command to device."""
+        async with self.state_lock:
+            await self.hass.async_add_executor_job(
+                self.braviarc.volume_down, self.audio_output
+            )
+            await self.async_request_refresh()
+
+    async def async_volume_mute(self, mute):
+        """Send mute command to device."""
+        async with self.state_lock:
+            await self.hass.async_add_executor_job(self.braviarc.mute_volume, mute)
+            await self.async_request_refresh()
+
+    async def async_media_play(self):
+        """Send play command to device."""
+        async with self.state_lock:
+            await self.hass.async_add_executor_job(self.braviarc.media_play)
+            self.playing = True
+            await self.async_request_refresh()
+
+    async def async_media_pause(self):
+        """Send pause command to device."""
+        async with self.state_lock:
+            await self.hass.async_add_executor_job(self.braviarc.media_pause)
+            self.playing = False
+            await self.async_request_refresh()
+
+    async def async_media_stop(self):
+        """Send stop command to device."""
+        async with self.state_lock:
+            await self.hass.async_add_executor_job(self.braviarc.media_stop)
+            self.playing = False
+            await self.async_request_refresh()
+
+    async def async_media_play_pause(self):
+        """Simulate play/pause command."""
+        if self.playing:
+            await self.async_media_pause()
+        else:
+            await self.async_media_play()
+
```

Comment:
We can remove this. It's already supported by the base media player entity.
https://github.com/home-assistant/core/blob/aaf3a5a9c5940df486d18dba894f2a43d6d3fcd4/homeassistant/components/media_player/__init__.py#L840-L850

```suggestion
```